api/utils.py

```diff
@@ -127,220 +128,299 @@ async def __call__(self, request: Request, security_scopes: SecurityScopes, retu
-        item = await query.where(orm_model.id == model_id).gino.first()
-        if custom_methods.get("get_one"):
-            item = await custom_methods["get_one"](model_id, user, item, internal)
+    @property
+    def paths(self) -> Dict[str, str]:
+        item_path = path_join(self.path, "{model_id}")
+        batch_path = path_join(self.path, "batch")
+        count_path = path_join(self.path, "count")
+        return {
+            "get": self.path,
+            "get_count": count_path,
+            "get_one": item_path,
+            "post": self.path,
+            "put": item_path,
+            "patch": item_path,
+            "delete": item_path,
+            "batch_action": batch_path,
+        }
+
+    @property
+    def response_models(self) -> Dict[str, Type]:
+        display_model = self.pydantic_model if not self.display_model else self.display_model
+        pagination_response = create_pydantic_model(
+            f"PaginationResponse_{display_model.__name__}",
+            count=(int, ...),
+            next=(Optional[str], None),
+            previous=(Optional[str], None),
+            result=(List[display_model], ...),
+            __base__=BaseModel,
+        )
+        return {
+            "get": pagination_response,
+            "get_count": int,
+            "get_one": display_model if self.get_one_model else None,
+            "post": display_model,
+            "put": display_model,
+            "patch": display_model,
+            "delete": display_model,
+        }
+
+    async def _get_one(self, model_id: int, user: schemes.User, internal: bool = False):
+        query = self.orm_model.query
+        if self.orm_model != models.User and user:
+            query = query.where(self.orm_model.user_id == user.id)
+        item = await query.where(self.orm_model.id == model_id).gino.first()
+        if self.custom_methods.get("get_one"):
+            item = await self.custom_methods["get_one"](model_id, user, item, internal)
         if not item:
             raise HTTPException(status_code=404, detail=f"Object with id {model_id} does not exist!")
         return item
 
-    async def get(
-        pagination: pagination.Pagination = Depends(),
-        user: Union[None, schemes.User] = Security(auth_dependency, scopes=scopes["get_all"]),
-    ):
-        if custom_methods.get("get"):
-            return await custom_methods["get"](pagination, user)
-        else:
-            return await pagination.paginate(orm_model, user.id)
-
-    async def get_count(user: Union[None, schemes.User] = Security(auth_dependency, scopes=scopes["get_count"])):
-        return (
-            await (
-                (orm_model.query.where(orm_model.user_id == user.id) if orm_model != models.User else orm_model.query)
-                .with_only_columns([db.db.func.count(distinct(orm_model.id))])
-                .order_by(None)
-                .gino.scalar()
+    def _get(self):
```

Comment:
Not sure why the code below doesn't work:
```python
async def get(pagination: pagination.Pagination = Depends(), user: Union[None, schemes.User] = None):
    user = user if user else Security(self.auth_dependency, scopes=self.scopes["get_all"])
```

Reply:
That's why I have to put it into a wrapper function

Reply:
You could use
```
try:
     user = await self.auth_dependency(request, SecurityScopes(self.scopes["get_all"]))
except HTTPException:
    user = None
```
But actually using a wrapper is cleaner

Reply:
I see. Thanks